src/shogun/base/SGObject.cpp

```diff
@@ -1060,6 +1060,49 @@ void CSGObject::put(const std::string& name, CSGObject* value)
 	    self->map[BaseTag(name)].get_value().type().c_str());
 }
 
+void CSGObject::add(const std::string& name, CSGObject* value)
+{
+	REQUIRE(
+	    value, "Cannot add to %s::%s, no object provided.\n", get_name(),
+	    name.c_str());
+
+	Tag<CDynamicObjectArray*> tag_array_sg(name);
+	if (has(tag_array_sg))
+	{
+		auto array = get(tag_array_sg);
+		if (!array)
+		{
+			SG_ERROR(
+			    "Cannot add object %s to parameters %s::%s of type %s, array "
+			    "is not instantiated.\n",
+			    value->get_name(), get_name(), name.c_str(),
+			    self->map[BaseTag(name)].get_value().type().c_str());
+		}
+
+		array->push_back(value);
+		return;
+	}
+
+	Tag<std::vector<CKernel*>> tag_array_kernel(name);
```

Comment:
and what happens if it's std::vector<FOO>? i mean the idea here is that you dispatch into whatever we support?

Reply:
Yes exactly, this is inherited from the design of put.

Reply:
for the record: i dont like it... and its even worse than the put :)

Reply:
these type of dynamic dispatchers, along with the ones that are actually being generated by swig should go to the swig interface itself. i mean in some way or other we are just re-implementing here things *manually* that are actually being automagically generated by SWIG - see dynamic dispatching depending on function args.